src/app/public/modules/shared/initialization.service.ts

```diff
@@ -6,13 +6,38 @@ import {
   BBHelpClient
 } from '@blackbaud/help-client';
 
+import {
+  SkyAppWindowRef
+} from '@skyux/core';
+
+import {
+  SkyAppConfig
+} from '@skyux/config';
+
 import {
   HelpWidgetConfig
 } from './widget-config';
 
 @Injectable()
 export class HelpInitializationService {
-  public load(config: HelpWidgetConfig) {
-      return BBHelpClient.load(config);
+  public constructor(private windowRef: SkyAppWindowRef, private config: SkyAppConfig) {
+  }
+
+  public load(config: HelpWidgetConfig = {}) {
+    const extraConfig: HelpWidgetConfig = {};
+    if (this.config.runtime.params.has('svcid')) {
+      extraConfig.extends = this.config.runtime.params.get('svcid');
+    }
+
+    if (this.config.runtime.params.has('envid')) {
+      config.environmentId = this.config.runtime.params.get('envid');
+    }
+
+    const skyuxHost = this.windowRef.nativeWindow.SKYUX_HOST;
+    if (skyuxHost && !config.locale) {
+      const languages = skyuxHost.acceptLanguage || '';
+      extraConfig.locale = languages.split(',')[0];
+    }
+    return BBHelpClient.load({...config, ...extraConfig});
```

Comment:
Object manipulation is so tricky in JS.  See https://scotch.io/bar-talk/copying-objects-in-javascript for all the pitfalls with each technique.  For example, the spread operator will only do a shallow copy.  That's not currently a problem for the `HelpWidgetConfig` type, but I don't know if that will always be true.

I know in the original code that this was copied from, we manipulated the original object.  Good or bad, we may be inadvertently introducing a breaking change for code that's relying on those properties to be set.

I'd be up for further discussion, but I'm inclined to leave the code as it was and manipulate the `config` object passed in.

Reply:
that's fair. will update.

Reply:
actually, im not sure how i feel about this. take the following

```javascript
function foo(param){
  param.bar = 'i changed yo param';
  // do stuff
}

let x = { bar: 'the original bar' };
console.log(x.bar);   // the original bar
foo(x);
console.log(x.bar);  // i changed yo param
```

i'm not a fan of functions that have side effects like this in general. but i won't fight the philosophical fight. 

more tangibly and pressing, if i'm changing this config object, is there any impact to builder? if i've messed with something it has passed me w/o it knowing will it mess something up?

Reply:
I totally agree.  It's a poorly written function to begin with.  I'll take half the credit for that and half belongs to JS making it extremely hard to do a deep copy, while maintain circular references and methods (anything that can't be converted to JSON).

What I'm saying is the current implementation is altering the `skyAppConfig.skyux.help` object.  So if something read `skyAppConfig.skyux.help` before https://github.com/blackbaud/skyux-sdk-builder/blob/a1331cd7a44554e34b0f9cef6109efb4dffa9e4c/src/app/app.component.ts#L296-L306 was executed, it could have different values if being read after (exactly like the example you posted above).

My fear is that we're relying on those values altered by this block of code to be set later in the app lifecycle, for example when the help widget is invoked.

Reply:
yea, i c. j/ keep existing behavior seems like a good strategy.

> half belongs to JS making it extremely hard to do a deep copy, while maintain circular references and methods

agreed.